### ISSUE TITLE
Should the docs be in this repository or in the website?

### ISSUE THREAD

**Issue by tomleah** [appsody/docs#54]
## Proposal 1 (Today):
- All documentation is found within `/docs` in the docs repository [here](https://github.com/appsody/docs/tree/master/docs).
- The plan for the website is that on deploy, the website will pull in all documentation from here and render it. We will use some sort of [build process](https://github.com/appsody/website/issues/48) to handle pulling in the docs.
- To preview documentation the user will need to clone both the docs and website repository to check their changes. It will also require some modification to pull in the new documentation.
- When a pull request is merged in docs, the website will need to be redeployed to add the changes.

## Proposal 2: 
- Documentation is moved to the website repository under `content/docs`.
- Users will only need to clone 1 repository to add and view documentation.
- There is no need for a build system to bring in documents from another location.
- The website will still need to be redeployed when a doc is merged.

**Comment by tomleah** [comment]
Proposal 1 has a lot more overhead and is more complicated, build process and contributor experience wise. 

Proposal 2 is simpler but we lose the separation between docs and website content. A concern is that is it intuitive for users to look in the website repo if they want to contribute documentation.  

We should consider what the project needs now and what it needs in the future. I would like to understand if we see the need for documentation versioning and the best way to solve this. After a conversation with @ddunn2 we could keep the latest docs in the website and older versions archived in the docs repo.

**Comment by kylegc** [comment]
What if we instead pull the website into this docs repo? That would simplify things because there's only one repo, but also keep the intuition for contributors (and users) to look for docs here. The website would be separated in a sub-directory of the docs repo. You can consider the entire website as documentation so I don't think this is too far fetched.

**Comment by tomleah** [comment]
I think there is going to be pros and cons, *if* the 2 repos are merged, to what the final location is. Overall I think more people will be contributing to docs rather than the website so merging into docs repo makes more sense to me in the case where we are concerned about discoverability. 

A problem might come when/if we start adding blogs or tutorials to the website (see https://github.com/appsody/website/issues/55) and how appsody content, in general, is stored/located.

**Comment by neeraj-laad** [comment]
I like the separation that **Proposal 1** offers, but don't like the fact that user's will have to clone both repositories before they can test/preview their changes. 

I would say we should merge them into a single repository, and keep the contributor experience and build process as simple as possible. If on a future date we have a reason to separate them out, we can consider it at that point rather than making process complicated today.

I also like suggestion from @kylegc to call they merged repository docs. Users will know where to go to contribute changes and they are more likely to make changes to docs vs website.

**Comment by neeraj-laad** [comment]
Another thought that comes to my mind is to have the docs as a separate app perhaps published on on `docs.appsody.dev` and publish those from `docs` repository. That way it is independent of the website, and can have independent repository, build process, release process etc.

The only downside is there might be duplication of some styles etc fi we want to use those across the website and docs.

**Comment by mhamwala** [comment]
I really like the separation between docs and website. I think that merging to two would make for a worse contributor experience overall, in terms of new contributors creating issues and discoverability of each docs/website.

It might be a few steps right now but I think that we can create clear documentation for how a user can locally view their docs change.

Like @tomleah said I think merging will cause issues later on in terms of when we address this issue https://github.com/appsody/website/issues/55). The singular repo might become overpopulated with website/docs/tutorial issues and pull requests.

To summarize I prefer proposal 1.

**Comment by sympatheticmoose** [comment]
Similarly I like the separation... I feel with a decent style guide for doc contributions we can hopefully limit the need for cloning/locally deploying the website to check how it looks/acts.

**Comment by sympatheticmoose** [comment]
Interesting looking at interesting looking at https://www.gatsbyjs.org/contributing/docs-contributions/#options-for-contributing-to-the-gatsby-docs (and the fact that they have the website as a `www` folder in the gatsbyjs/gatsby repo). They suggest Github UI for simple changes and cloning/running locally for more comprehensive

**Comment by ddunn2** [comment]
So with the separation of repos, one for docs and one for the website, to contribute a change I would need to: 
1. Fork the docs repo & Clone my fork
2. Add my new document in the correct format
3. Push my changes
4. Clone down the website repo
5. Update the website config to point at where ever my draft version of the doc is
6. I can now preview the doc in the website to ensure everything is working

Is that correct? 
If so that's not exactly an 'quick' process in my opinion, assuming I have the above steps correct. Unless you're going to remove the need to preview in the website and just preview as GitHub Markdown? In this way you'd need to be able to spot whether the Frontmatter is correct yourselves and also consider that Gatsby and GitHub don't always render Markdown in the same way as seen here: https://github.com/appsody/website/issues/42

As an aside for me, the logical place to update documentation would be the place I'm viewing it. 

For Kitura, I've taken the approach of keeping the docs within the website but in a separate directory from the website code. I have a `/src` directory for the website code and a `/content` directory for documentation. So separation is maintained, just without having another repo to do it. You can see that [here](https://github.com/ddunn2/kitura-gatsby)

In this way to contribute a doc you would: 
1. Clone down website repo
2. Add the new document in the correct format
3. Preview the doc in the website to ensure everything is working

To me, it seems like separating the repos would be more work to setup the pipelining and a poorer experience for anyone wanting to contribute docs. Again I may have the steps involved wrong, so feel free to correct me if I did 😅

**Comment by neeraj-laad** [comment]
@tomleah @ddunn2  I'm continuing to think more on the lines of https://github.com/appsody/docs/issues/54#issuecomment-508100889

Could we have the docs as a separate Gatsby app that is stored, managed, built and deployed all from the docs repo.. effectively making it independent from the website. 

It also simplifies the publishing side of things:
- no more fetching files from docs repo while building the website repo.
- no more redeploying the website when a doc has changed and needs to be deployed.

Ofcourse, an important aspect would be to understand how much are they dependent on each other and what would how much we would have to duplicate or have in both repos to make this happen?

**Comment by tomleah** [comment]
@neeraj-laad the problem with having 2 separate Gatsby apps is that the react components and all the site's styles would be duplicated and a change to the navbar, for example, would need to happen in 2 places to keep the overall site consistent.

**Comment by tomleah** [comment]
Another option is to have the separated repositories and provide a script that would clone the website repository, move the docs into the right place and then host the website for documentation contributors. Speaking to @BethGriggs this could be hidden behind an alias so that the user isn't exposed to the script directly.

**Comment by tomleah** [comment]
In the interest of time as this has been open a week, the 2 repositories will be **merged**. This will keep the contributor experience and the build process simple.

Going forward:
- The docs repository will remain with a `www` folder which will include the website.
- Issues will be transferred to the docs repo with a `website` label attached.
- The website repo will be archived

**Comment by tomleah** [comment]
Merged in https://github.com/appsody/website/pull/63
